gym_ignition/base/robot/robot_joints.py

```diff
@@ -0,0 +1,361 @@
+    def joint_velocity(self, joint_name: str) -> float:
+        """
+        Return the generalized velocity of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+
+        Returns:
+            The joint generalized velocity.
+
+        """
+
+    @abstractmethod
+    def joint_positions(self) -> np.ndarray:
+        """
+        Return the generalized positions of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized positions.
+        """
+
+    @abstractmethod
+    def joint_velocities(self) -> np.ndarray:
+        """
+        Return the generalized velocities of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized velocities.
+        """
+
+    @abstractmethod
+    def joint_pid(self, joint_name: str) -> Union[PID, None]:
+        """
+        Return the PID associated to the specified joint.
+
+        Args:
+            joint_name: The joint name.
+
+        Returns:
+            The low-level joint PID or None if no PID was configured.
+        """
+
+    @abstractmethod
+    def dt(self) -> float:
+        """
+        Returns the step_size of the low-level PID controllers.
+
+        Returns:
+            The PIDs step_size.
+        """
+
+    @abstractmethod
+    def set_dt(self, step_size: float) -> bool:
+        """
+        Set the step size used by the low-level PIDs.
+
+        Args:
+            step_size: The PIDs step size.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_position_target(self, joint_name: str, position_reference: float) -> \
+            bool:
+        """
+        Set the position reference of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            position_reference: The joint position reference.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_velocity_target(self, joint_name: str, velocity_reference: float) -> \
+            bool:
+        """
+        Set the velocity reference of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            velocity_reference: The joint velocity reference.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_force(self, joint_name: str, force: float, clip: bool = False) -> bool:
+        """
+        Set the generalized force applied to the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            force: The joint generalize force.
+            clip: Clip joint forces reading effort limits.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_position(self, joint_name: str, position: float) -> bool:
```

Comment:
I think that is may be a bit confusing that `set_base_pose` changes the state of the simulation, while `set_joint_position` is setting a setpoint of a low-level PID controller.

Reply:
> For what concern naming, you're right. There's a mismatch with the baseframe interface. Maybe `reset_base_pose` would be more appropriate.

:+1:

Reply:
I renamed `set_base_{pose,velocity}` to `reset_base_{pose,velocity}`.